vnext/Microsoft.ReactNative/Modules/ImageViewManagerModule.cpp

```diff
@@ -77,96 +58,67 @@ winrt::fire_and_forget GetImageSizeAsync(
 
-void ImageViewManagerModule::ImageViewManagerModuleImpl::getSize(
-    std::string uri,
-    Callback successCallback,
-    Callback errorCallback) {
-  folly::dynamic headers{};
-  GetImageSizeAsync(uri, headers, successCallback, errorCallback);
-}
-
-void ImageViewManagerModule::ImageViewManagerModuleImpl::getSizeWithHeaders(
-    std::string uri,
-    folly::dynamic &headers,
-    Callback successCallback,
-    Callback errorCallback) {
-  GetImageSizeAsync(uri, headers, successCallback, errorCallback);
+void ImageLoader::Initialize(React::ReactContext const &reactContext) noexcept {
+  m_context = reactContext;
 }
 
-void ImageViewManagerModule::ImageViewManagerModuleImpl::prefetchImage(
-    std::string /*uri*/,
-    Callback successCallback,
-    Callback /*errorCallback*/) {
-  // NotYetImplemented
-  successCallback({});
+void ImageLoader::getSize(std::string uri, React::ReactPromise<React::JSValue> &&result) noexcept {
+  getSizeWithHeaders(std::move(uri), {}, std::move(result));
 }
 
-void ImageViewManagerModule::ImageViewManagerModuleImpl::queryCache(
-    const folly::dynamic & /*requests*/,
-    Callback successCallback,
-    Callback /*errorCallback*/) {
-  // NotYetImplemented
-  successCallback({folly::dynamic::object()});
+void ImageLoader::getSizeWithHeaders(
+    std::string uri,
+    React::JSValue &&headers,
+    React::ReactPromise<React::JSValue> &&result) noexcept {
+  m_context.UIDispatcher().Post([context = m_context,
+                                 uri = std::move(uri),
+                                 headers = std::move(headers),
+                                 result = std::move(result)]() mutable noexcept {
+    GetImageSizeAsync(
+        std::move(uri),
+        std::move(headers),
+        [result, context](int32_t width, int32_t height) noexcept {
+          context.JSDispatcher().Post([result = std::move(result), width, height]() noexcept {
+            result.Resolve(React::JSValueArray{width, height});
+          });
+        },
+        [result, context]() noexcept {
+          context.JSDispatcher().Post([result = std::move(result)]() noexcept { result.Reject("Failed"); });
+        });
+  });
 }
 
-//
-// ImageViewManagerModule
-//
-const char *ImageViewManagerModule::name = "ImageLoader";
-
-ImageViewManagerModule::ImageViewManagerModule()
-    : m_imageViewManagerModule(std::make_shared<ImageViewManagerModuleImpl>(this)) {}
-
-ImageViewManagerModule::~ImageViewManagerModule() {}
-
-std::string ImageViewManagerModule::getName() {
-  return name;
+void ImageLoader::prefetchImage(std::string uri, React::ReactPromise<React::JSValue> &&result) noexcept {
+  // NYI
+  result.Resolve(true);
 }
 
-std::map<std::string, folly::dynamic> ImageViewManagerModule::getConstants() {
-  return {{}};
+void ImageLoader::prefetchImageWithMetadata(
+    std::string uri,
+    std::string queryRootName,
+    double rootTag,
+    React::ReactPromise<React::JSValue> &&result) noexcept {
+  // NYI
+  result.Resolve(true);
```

Comment:
I saw that we resolved before, but would rejecting be more appropriate here?

Reply:
Same for other functions

Reply:
Possibly, but I dont want to conflate moving this to the more type safe / efficient turbomodule and changing its functionality.
Resolving is certainly safer, as it'll claim that the image is ready to go.  Where apps that are assuming this API is working are much more likely to break if it was rejected.